hack/choco/update-choco-package.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+ 
+# Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+ 
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+ 
+version="${1:?TCE version argument empty. Example usage: ./hack/choco/update-choco-package.sh v0.10.0}"
+: "${GITHUB_TOKEN:?GITHUB_TOKEN is not set}"
+ 
+ 
+temp_dir=$(mktemp -d)
+ 
+pushd "${temp_dir}"
+
+TCE_REPO="https://github.com/vmware-tanzu/community-edition/" 
+TCE_REPO_RELEASES_URL="https://github.com/vmware-tanzu/community-edition/releases"
+TCE_WINDOWS_TAR_BALL_FILE="tce-darwin-amd64-${version}.tar.gz"
+TCE_CHECKSUMS_FILE="tce-checksums.txt"
+ 
+echo "Checking if the necessary files exist for the TCE ${version} release"
+ 
+curl -f -I -L \
+   "${TCE_REPO_RELEASES_URL}/download/${version}/${TCE_WINDOWS_TAR_BALL_FILE}" > /dev/null || {
+       echo "${TCE_WINDOWS_TAR_BALL_FILE} is not accessible in TCE ${version} release"
+       exit 1
+   }
+ 
+wget "${TCE_REPO_RELEASES_URL}/download/${version}/${TCE_CHECKSUMS_FILE}" || {
+   echo "${TCE_CHECKSUMS_FILE} is not accessible in TCE ${version} release"
+   exit 1
+}
+ 
+git clone "${TCE_REPO}"
+
+cd community-edition
+
+# make sure we are on main branch before checking out
+git checkout main
+ 
+PR_BRANCH="update-tce-to-${version}-${RANDOM}"
+ 
+# Random number in branch name in case there's already some branch for the version update,
+# though there shouldn't be one. There could be one if the other branch's PR tests failed and didn't merge
+git checkout -b "${PR_BRANCH}"
+ 
+# Replacing old version with the latest stable released version.
+# Using -i so that it works on Mac and Linux OS, so that it's useful for local development.
+sed -i -e "s/\(\$releaseVersion =\).*/\$releaseVersion = ""'${version}'""/g" hack/choco/tools/chocolateyinstall.ps1
+rm -fv hack/choco/tools/chocolateyinstall.ps1-e
+ 
+version="${version:1}"
+sed -i -e "s/\(<version>\).*\(<\/version>\)/<version>""${version}""\<\/version>/g" hack/choco/tanzu-community-edition.nuspec
```

Comment:
A general question - have you tried running the ps1 script after it's been modified by this script? I'm wondering if the newline differences between unix and windows will arise due to this editing.

Reply:
I haven’t tried but I am planning a upgrade test for [#2231](https://github.com/vmware-tanzu/community-edition/issues/2231)after this script in which i will test once this script will update the version in ps1 script.